broadcast/src/buffered/engine.rs

```diff
@@ -352,9 +358,16 @@ impl<
     /// Respond to a waiter with an optional message.
     /// Increments the appropriate metric based on the result.
     fn respond_get(&mut self, responder: oneshot::Sender<Option<M>>, msg: Option<M>) {
+        let found = msg.is_some();
         let result = responder.send(msg);
         self.metrics.get.inc(match result {
-            Ok(_) => Status::Success,
+            Ok(_) => {
+                if found {
+                    Status::Success
+                } else {
+                    Status::Failure
+                }
+            }
```

Comment:
nit: ```suggestion
            Ok(_) if found => Status::Success,
            Ok(_) => Status::Failure
```

I might also suggest defining `let status = match result {...}` outside of `self.metrics.get.inc(...)` for additional readability

Reply:
woah, that's a new trick lol